rinja_parser/src/node.rs

```diff
@@ -1057,6 +1057,25 @@ impl<'a> Comment<'a> {
 #[derive(Clone, Copy, Debug, PartialEq)]
 pub struct Ws(pub Option<Whitespace>, pub Option<Whitespace>);
 
+fn end_node<'a, 'g: 'a>(
+    node: &'g str,
+    expected: &'g str,
+) -> impl Fn(&'a str) -> ParseResult<'a> + 'g {
+    move |start| {
+        let (i, actual) = ws(identifier)(start)?;
+        if actual == expected {
+            Ok((i, actual))
+        } else if actual == "end" {
+            Err(nom::Err::Failure(ErrorContext::new(
+                format!("`{node}` nodes are terminated with `{expected}`"),
```

Comment:
I would reword it a bit to make it more obvious that the "ender" is missing:

```suggestion
                format!("missing `{expected}` to terminate `{node}` node"),
```

Reply:
I thought my wording expresses better that `end` is not a keyword. Maybe the combined message would be best?

```rust
"\
missing `{expected}` to terminate `{node}` node
`{node}` nodes are terminated with `{expected}`, not `end`"
```

Reply:
I'm still not entirely sold on your message improvement. Also, thinking about this some more, I think the `else if actual == "end"` block should be merged with the `else`.

Message suggestion v3:

```
expected `{expected}` to terminate `{node}` node, found `{actual}`
```

Reply:
> I think the `else if actual == "end"` block should be merged with the else.

Not possible. `fail()` returns an recoverable error, but for `actual == "end"` we return an unrecoverable failure. (The name `fail()` might not have been the best choice.)

Reply:
Oh I see. We tell that it looks to be looking further because it doesn't match the need here. Hum... Did you encounter this problem yourself? (using `end` instead of the right keyword)

Reply:
I searched what usage questions users were having about askama, and came across this thread: <https://users.rust-lang.org/t/how-to-solve-error-about-askama/59431>